dart-parser/src/parser.rs

```diff
@@ -1,56 +1,37 @@
 mod class;
+mod comment;
 mod common;
 mod directive;
 mod string;
 
 use std::str;
 
-use nom::{
-    branch::alt,
-    bytes::complete::{is_not, tag},
-    combinator::{eof, recognize},
-    multi::many0,
-    sequence::tuple,
-    IResult, Parser,
-};
+use nom::{branch::alt, combinator::eof, multi::many0, IResult, Parser};
 
-use crate::{
-    dart::*,
-    parser::{class::class, common::*},
-};
+use crate::{dart::*, parser::class::class};
 
-use self::directive::directive;
+use self::{comment::comments, directive::directive};
 
 pub fn parse(s: &str) -> IResult<&str, Vec<Dart>> {
     let (s, items) = many0(alt((
-        alt((spbr, comment)).map(Dart::Verbatim),
```

Comment:
You're not allowing "free" whitespace anymore (you extracted the `comment`, but completely dropped `spbr`), this is likely the reason why the old tests are failing

Reply:
Ah yes!